src/app/workflow/workflow.js

```diff
@@ -149,21 +148,35 @@ angular.module('app.workflow', [
   self.forms = forms;
   self.delete = function(event, workflow) {
     event.preventDefault();
-    var confirm = $mdDialog.confirm()
-          .title('Would you like to delete workflow #'+workflow.id+'?')
-          .textContent(workflow.title)
-          .ariaLabel('Delete workflow')
-          .targetEvent(event)
-          .ok('Proceed')
-          .cancel('Cancel');
-    $mdDialog.show(confirm).then(function() {
-      return workflowManager.delete(workflow)
-      .then(function() {
-        $state.go('app.workflow', null, {reload: true});
-      }, function(err) {
-        throw err;
+    var title,
+        confirm;
+    workorderManager.list().then(function(workorders){
+      var workorder = workorders.filter(function(workorder) {
+        return String(workorder.workflowId) === String(workflow.id);
       })
-    });
+      if (workorder.length) {
+        title = "Workflow is used at least by 1 workorder, are you sure you want to delete workflow #'"+workflow.id+"?";
+      }
+      else {
+        title = "Would you like to delete workflow #"+workflow.id+"?";
+      }
+      confirm = $mdDialog.confirm()
+            .title(title)
+            .textContent(workflow.title)
+            .ariaLabel('Delete workflow')
+            .targetEvent(event)
+            .ok('Proceed')
+            .cancel('Cancel');
+    }).then(function() {
+      $mdDialog.show(confirm).then(function() {
```

Comment:
If we un-nest the 3 nested promises here we will have a nicely chained promise with complete error handling.